Makefile

```diff
@@ -1,24 +1,46 @@
-.PHONY: setup update run
+# ------------------------------------------------------------------------------
+# Development environment setup:
+# ------------------------------------------------------------------------------
 
+setup: ./era-test-node setup-execution-helper setup-seaport
+
+# Clones the `era-test-node` into `./era-test-node/`.
+.PHONY: ./era-test-node
+./era-test-node: 
+	git clone --depth 1 git@github.com:matter-labs/era-test-node.git
+
+.PHONY: setup-execution-helper
 setup-execution-helper: 
 	cd ExecutionHelper && yarn install 
 
-compile-and-deploy-execution-helper:
-	cd ExecutionHelper && yarn hardhat compile && yarn hardhat deploy-zksync --script deploy.ts
-
+.PHONY: setup-seaport
 setup-seaport: 
 	yarn install
 
-compile-seaport:
-	yarn hardhat compile 
+# ------------------------------------------------------------------------------
+# Development environment update:
+# ------------------------------------------------------------------------------
 
-clone-node: 
-	git clone --depth 1 git@github.com:matter-labs/era-test-node.git
+.PHONY: update.era-test-node
+update.era-test-node: ./era-test-node
+	cd era-test-node && git pull
 
-setup: clone-node setup-execution-helper compile-and-deploy-execution-helper setup-seaport compile-seaport
+# ------------------------------------------------------------------------------
+# Compile:
+# ------------------------------------------------------------------------------
 
-update:
-	cd era-test-node && git pull
+.PHONY: compile-and-deploy-execution-helper
+compile-and-deploy-execution-helper:
+	cd ExecutionHelper && yarn hardhat compile && yarn hardhat deploy-zksync --script deploy.ts
+
+.PHONY: compile-seaport
+compile-seaport: compile-and-deploy-execution-helper
+	yarn hardhat compile 
+
+# ------------------------------------------------------------------------------
+# Developer tools:
+# ------------------------------------------------------------------------------
 
-run:
+.PHONY: run-era-test-node
+run-era-test-node: ./era-test-node
```

Comment:
This fails if the directory is already cloned

Reply:
A suggestion is to change the era-test-node target to this:
```makefile
./era-test-node: 
	if [ ! -d "era-test-node" ]; then \
	echo "Cloning era test node since it's missing..."; \
	git clone --depth 1 git@github.com:matter-labs/era-test-node.git; \
	fi 

```

Reply:
No, it does not because it is not a `.PHONY` target. Try calling it repeatedly to verify that it's idempotent.